pkg/kubelet/nodestatus/setters.go

```diff
@@ -57,25 +57,46 @@ const (
 type Setter func(node *v1.Node) error
 
 // NodeAddress returns a Setter that updates address-related information on the node.
-func NodeAddress(nodeIP net.IP, // typically Kubelet.nodeIP
+func NodeAddress(nodeIPs []net.IP, // typically Kubelet.nodeIPs
 	validateNodeIPFunc func(net.IP) error, // typically Kubelet.nodeIPValidator
 	hostname string, // typically Kubelet.hostname
 	hostnameOverridden bool, // was the hostname force set?
 	externalCloudProvider bool, // typically Kubelet.externalCloudProvider
 	cloud cloudprovider.Interface, // typically Kubelet.cloud
 	nodeAddressesFunc func() ([]v1.NodeAddress, error), // typically Kubelet.cloudResourceSyncManager.NodeAddresses
 ) Setter {
+	var nodeIP, secondaryNodeIP net.IP
+	if len(nodeIPs) > 0 {
+		nodeIP = nodeIPs[0]
+	}
 	preferIPv4 := nodeIP == nil || nodeIP.To4() != nil
 	isPreferredIPFamily := func(ip net.IP) bool { return (ip.To4() != nil) == preferIPv4 }
 	nodeIPSpecified := nodeIP != nil && !nodeIP.IsUnspecified()
 
+	if len(nodeIPs) > 1 {
+		if externalCloudProvider || cloud != nil {
+			klog.Warningf("Cloud providers do not support multiple --node-ip values; ignoring second value")
+		} else if nodeIP.IsUnspecified() || secondaryNodeIP.IsUnspecified() {
+			klog.Warningf("Multi-valued --node-ip cannot include 0.0.0.0 or ::; ignoring second value")
```

Comment:
this needs to move to option validation in option file. if we are to resolve the cloud comment earlier then we need to leave this validation here.